libs/ts-rest/core/src/lib/client.ts

```diff
@@ -67,6 +67,20 @@ export type ApiRouteResponse<T> =
       body: unknown;
     };
 
+export type ApiResponseForRoute<T extends AppRoute> =
+  | {
+      [K in keyof T['responses']]: {
+        status: K
+        body: ZodInferOrType<T['responses'][K]>
+      }
+    }[keyof T['responses']]
+  & {
+      status: Exclude<HTTPStatusCode, keyof T>
+      body: unknown
+    };
```

Comment:
You could simplify this into

```suggestion
export type ApiResponseForRoute<T extends AppRoute> = ApiRouteResponse<T['responses']>;
```

Also an intersection type would cause this to resolve into `never`

Reply:
@Gabrola so I tried both types (intersection and union). Let's take an example from `getPosts` contract schema: 
```
    responses: {
      200: z.object({
        posts: PostSchema.array(),
        count: z.number(),
        skip: z.number(),
        take: z.number(),
      }),
    },
```
when using the `union` type, if I were to add an extra key to this response object like so:

✅ 
```
    return {
      status: 200,
      body: { posts, count: totalPosts, skip, take, extraParam: 'hello' },
    };
```
I do not receive a TS error. Using the `intersection` type will result in an error:

```
    return {
      status: 200,
      body: { posts, count: totalPosts, skip, take, extraParam: 'hello'  <--- ❌  },
    };
```

Maybe we should discuss the intended functionality, but in my opinion, the return types should be strict: "Strict" meaning we don't allow extra properties in response object other than what's defined in the contract. Also, if properties are missing that are required, we should also enforce a TS error.

Reply:
The reason this is happening is because you have a slight bug in your code. The second part was causing it to resolve to a `body: unknown` even for status 200, so you were ending up with a body type of `body: z.infer<...> | unknown` which is why it was letting through extra properties. If you fix that bug, an intersection type would resolve to never because for example, there is no intersection between `{ status: 200; body: any } & { status: 400; body unknown }`, so you end up with `never`

```suggestion
export type ApiResponseForRoute<T extends AppRoute> =
  | {
      [K in keyof T['responses']]: {
        status: K
        body: ZodInferOrType<T['responses'][K]>
      }
    }[keyof T['responses']]
  | {
      status: Exclude<HTTPStatusCode, keyof T['responses']>
      body: unknown
    };
```

Reply:
AH, you're totally right! Thank you! This is perfect 👏 a silly mistake on my end.

Reply:
So that's why `ApiRouteResponse<T['responses']>` should just work, since it's essentially the same and you're passing in that type

Reply:
AH, you're totally right! Thank you! This is perfect 👏 a silly mistake on my end.

Reply:
So that's why ApiRouteResponse<T['responses']> should just work, since it's essentially the same and you're passing in that type